shared/shared/postgresql_backend/sql_read_database_backend_service.py

```diff
@@ -1,30 +1,261 @@
+    ) -> Any:
+        query, arguments, sql_params = self.build_filter_query(
+            collection, filter, fields_params
+        )
+        value = self.connection.query(query, arguments, sql_params)
+        return value[0].copy()
+
+    def fetch_list_of_models(
+        self,
+        query: str,
+        arguments: List[str],
+        sql_parameters: List[str],
+        mapped_fields: List[str],
+    ) -> List[Model]:
+        result = self.connection.query(query, arguments, sql_parameters)
+        if len(mapped_fields) > 0:
+            models = [row.copy() for row in result]
+        else:
+            models = [row["data"] for row in result]
+        return models
+
+    def get_unique_mapped_fields(
+        self, mapped_fields_per_collection: Dict[str, List[str]]
+    ) -> List[str]:
+        if len(mapped_fields_per_collection):
+            unique_mapped_fields: Set[str] = set.union(
+                *[set(fields) for fields in mapped_fields_per_collection.values()]
+            )
+            return list(unique_mapped_fields)
+        else:
+            return []
+
+    def build_select_mapped_fields(
+        self,
+        unique_mapped_fields: List[str],
+        mapped_fields_per_collection: Dict[str, List[str]] = None,
+    ) -> str:
+        if len(unique_mapped_fields) == 0 or (
+            mapped_fields_per_collection
+            and self.mapped_fields_map_has_empty_entry(mapped_fields_per_collection)
+        ):
+            # at least one collection needs all fields, so we just select data and
+            # calculate the mapped_fields later
+            return "data"
+        else:
+            return ", ".join(["data->%s AS {}"] * len(unique_mapped_fields))
+
+    def mapped_fields_map_has_empty_entry(
+        self, mapped_fields_per_collection: Dict[str, List[str]]
+    ) -> bool:
+        return not len(mapped_fields_per_collection) or any(
+            len(fields) == 0 for fields in mapped_fields_per_collection.values()
+        )
+
+    def build_models_from_result(
+        self, result, mapped_fields_per_collection: Dict[str, List[str]]
+    ) -> Dict[str, Model]:
+        result_map = {}
+        for row in result:
+            fqid = row["fqid"]
+            collection = collection_from_fqid(fqid)
+
+            if self.mapped_fields_map_has_empty_entry(mapped_fields_per_collection):
+                # at least one collection needs all fields, so we just selected data
+                model = row["data"]
+            else:
+                model = row.copy()
+
+            if (
+                collection in mapped_fields_per_collection
+                and len(mapped_fields_per_collection[collection]) > 0
+            ):
+                for key in list(model.keys()):
+                    if key not in mapped_fields_per_collection[collection]:
+                        del model[key]
+            result_map[fqid] = model
+
+        return result_map
+
+    def build_filter_query(
```

Comment:
Maybe you can move some filter related methods into an own service, e.g. "FilterQueryBilder" or so. This would reduce the complexity in this class.